lib/src/switch/Switch.tsx

```diff
@@ -63,61 +62,64 @@ const DxcSwitch = ({
 
   return (
     <ThemeProvider theme={colorsTheme.switch}>
-      <SwitchContainer margin={margin} size={size} onKeyDown={handleOnKeyDown}>
-        {labelPosition === "before" && hasLabel && (
+      <SwitchContainer
+        margin={margin}
+        size={size}
+        onKeyDown={handleOnKeyDown}
+        onClick={!disabled ? handlerSwitchChange : undefined}
+        disabled={disabled}
+      >
+        {labelPosition === "before" && label && (
           <LabelContainer
             id={labelId}
             labelPosition={labelPosition}
-            onClick={!disabled ? handlerSwitchChange : undefined}
             disabled={disabled}
             backgroundType={backgroundType}
-            hasLabel={hasLabel}
+            label={label}
           >
             {label} {optional && <>{translatedLabels.formFields.optionalLabel}</>}
           </LabelContainer>
         )}
-        <SwitchBase
-          labelPosition={labelPosition}
-          hasLabel={hasLabel}
-          htmlFor={labelId}
-          onClick={disabled === true ? () => {} : handlerSwitchChange}
-        >
+        <SwitchBase htmlFor={labelId}>
```

Comment:
`htmlFor={labelId}` is unnecessary. When the labelled element is inside the <label> they are already associated, so they are accessible.

Reply:
What do you mean by inside? `ValueInput` and `LabelContainer` are siblings in the DOM.

Reply:
`htmlFor` is used to match an input with its label/output. First, in this case, the `SwitchBase` is a container, so it doesn't need a `for` attribute. The relation is between <LabelContainer> and <DisabledSwitchTrack/SwitchTrack>, which is properly established by using the `aria-labelledBy`.

What I meant with inside is that the SwitchBase is a <label> and is associated automatically with its inner content (look at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#:~:text=To%20associate%20the%20with,same%20as%20the%20input's%20id%20)).

Reply:
Removed.